src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/osd/osd-form/osd-form.component.spec.ts

```diff
@@ -50,6 +55,45 @@ describe('OsdFormComponent', () => {
     }
   ];
 
+  const deploymentOptions: DeploymentOptions = {
+    options: {
+      cost_capacity: {
+        name: OsdDeploymentOptions.COST_CAPACITY,
+        available: true,
+        capacity: 0,
+        used: 0,
+        hdd_used: 0,
+        ssd_used: 0,
+        nvme_used: 0,
+        option_name: 'Cost/Capacity-optimized',
+        msg: 'All the available HDDs are selected'
+      },
```

Comment:
All the things  mentioned for the back-end data structs apply here, and they're more evident since most of these fields are not needed in the UI (ssd_used?), only the recommended option, right? That means that we're coupling in the back-end 2 or more different data classes.

The class we'd need here is what, at back-end, I defined as:
```py
# Named tuples allows us to quickly prototype immutable objects with initialization, defaults and asdict()
class OsdDeploymentOption(NamedTuple):
  id: str
  title: str
  desc: str
```
If we need some extra data (like total capacity, or number of disks), we could append some "hints" to this class (e.g.: the `getDeploymentRecommendation()` service returns the recommended DeploymentOption + extras: available + capacity + etc.).

Reply:
I think we decided on this in the previous PR https://github.com/ceph/ceph/pull/45190, and fields like hdd_used etc are needed on the later iterations when we try to fill the pie chart in the frontend (to visualize the storage used infos).